akka-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala

```diff
@@ -10,10 +10,12 @@ import akka.parboiled2._
 import akka.shapeless._
 import akka.http.scaladsl.model._
 
+private[http] case class HeaderParserSettings(uriParsingMode: Uri.ParsingMode = Uri.ParsingMode.Relaxed)
```

Comment:
I changed the Settings into a trait similar to `HttpHeaderParser.Settings` which simplified things and also introduced a val with the default settings.